sklearn/utils/estimator_checks.py

```diff
@@ -149,6 +150,32 @@ def check_estimator_sparse_data(name, Estimator):
         raise
 
 
+def check_dtype_object(name, Estimator):
```

Comment:
I'm not sure what you mean by "numeric features must be accepted" That is something that all tests require, right?
I would formulate it as "object dtype should be handled as numeric"